recipe/meta.yaml

```diff
@@ -1,43 +1,65 @@
-{% set version = "5.2.2" %}
+{% set name = "superlu" %}
+{% set version = "5.3.0" %}
 
 
 package:
-  name: superlu
+  name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: superlu-{{ version }}.tar.gz
-  url: http://crd-legacy.lbl.gov/~xiaoye/SuperLU/superlu_{{ version }}.tar.gz
-  sha256: 470334a72ba637578e34057f46948495e601a5988a602604f5576367e606a28c
+  url: https://github.com/xiaoyeli/{{ name }}/archive/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 3e464afa77335de200aeb739074a11e96d9bef6d0b519950cfa6684c4be1f350
+  patches:
+    # mc64 is under a comercial license.
+    # Patch taken from https://src.fedoraproject.org/rpms/SuperLU/raw/f38/f/superlu-removemc64.patch
+    - patches/superlu-removemc64.patch
```

Comment:
Is it really better to just do it like fedora did?
Our solution was less invasive in that it prevented the [mc64 flag](https://github.com/xiaoyeli/superlu/blob/f3be414a05d205331420cbdae8514c15025e6b83/SRC/cgsisx.c#L549) to be set to true in the cgsisx.c driver. This also offers the benefit of obliterating the source file `mc64ad.c` from the build system.
@JeanChristopheMorinPerso, maybe you can share your thoughts.

Reply:
Basically, doing this the way fedora did I guess triggers the `mc64ad_` call to fail, which will then make the code fall back to non-mc64 code, as shown [here](https://github.com/xiaoyeli/superlu/blob/f3be414a05d205331420cbdae8514c15025e6b83/SRC/cgsisx.c#L556C6-L556C6).

Reply:
I see this patch is safer from legal perspective it does not only execute mc64 code but also does not contain in it in the binary which by itself can violate the license, see in theory if mc64 function is still there, the C/C++ extern declaration can be used and the function can be called from different place

Reply:
What @M-Waszkiewicz-Anaconda said. Since the algorithm is commercially licensed, we don't want it in our binaries. We can't redistribute commercially licensed software.